gateway/policy.go

```diff
@@ -186,14 +186,18 @@ func parsePoliciesFromRPC(list string) (map[string]user.Policy, error) {
 	policies := make(map[string]user.Policy, len(dbPolicyList))
 
 	for _, p := range dbPolicyList {
-		p.ID = p.MID.Hex()
-		policies[p.MID.Hex()] = p
+		id := p.MID.Hex()
+		if allowExplicit && p.ID != "" {
+			id = p.ID
+		}
+		p.ID = id
+		policies[id] = p
 	}
 
 	return policies, nil
 }
 
-func (gw *Gateway) LoadPoliciesFromRPC(orgId string) (map[string]user.Policy, error) {
+func (gw *Gateway) LoadPoliciesFromRPC(orgId string, allowExplicit bool) (map[string]user.Policy, error) {
```

Comment:
Lets remove this param, as we have access to the Gw object then inside the function we can do something like:

`allowExplicitv:= gw.GetConfig().Policies.AllowExplicitPolicyID`

Reply:
I did it this way to maintain coherence with the other LoadPolicies method, `LoadPoliciesFromDashboard(endpoint, secret string, allowExplicit bool) map[string]user.Policy`. 

I think it's cleaner and it's going to make easier to test different scenarios of `LoadPoliciesFromRPC ` without the need of changing gateway configuration and reloading each time.

Reply:
Ahhhh!! makes sense!